apps/skilavottord/ws/src/app/modules/recyclingRequest/recyclingRequest.service.ts

```diff
@@ -110,6 +120,31 @@ export class RecyclingRequestService {
     return res
   }
 
+  async findRecyclingRequestWithPermno(
+    permno: string,
+    user: User,
+  ): Promise<RecyclingRequestModel[]> {
+    this.logger.info(
+      `---- Starting findRecyclingRequestWithPermno for ${permno} ----`,
+    )
+    const carList = await this.samgongustofaService.getVehicleInformation(
+      user.nationalId,
+    )
+    if (
+      carList.find((car) => car.isRecyclable && car.permno === permno) ||
+      [Role.recyclingCompany, Role.developer].includes(user.role)
```

Comment:
Would like to see this split for readability:
```typescript
const car = carList.find((car) => car.isRecyclable && car.permno === permno)
const hasPermission = [Role.recyclingCompany, Role.developer].includes(user.role)

if (...)
```

But scrap that... the permno should be handled by the `samgongustofaService` .. and the role checks should be handled by the resolver.